pkg/diff/sql_generator.go

```diff
@@ -689,10 +700,35 @@ func (t *tableSQLVertexGenerator) Delete(table schema.Table) ([]Statement, error
 func (t *tableSQLVertexGenerator) Alter(diff tableDiff) ([]Statement, error) {
 	if diff.old.IsPartition() != diff.new.IsPartition() {
 		return nil, fmt.Errorf("changing a partition to no longer be a partition (or vice versa): %w", ErrNotImplemented)
-	} else if diff.new.IsPartition() {
-		return t.alterPartition(diff)
 	}
 
+	var stmts []Statement
+	if diff.new.IsPartition() {
+		alterPartitionStmts, err := t.alterPartition(diff)
+		if err != nil {
+			return nil, fmt.Errorf("altering partition: %w", err)
+		}
+		stmts = alterPartitionStmts
+	} else {
+		alterBaseTableStmts, err := t.alterBaseTable(diff)
+		if err != nil {
+			return nil, fmt.Errorf("altering base table: %w", err)
+		}
+		stmts = alterBaseTableStmts
+	}
+
+	if diff.old.ReplicaIdentity != diff.new.ReplicaIdentity {
```

Comment:
Previously, we had the alter statement generation completely separated. Altering replica identity is the same for both partitions and base tables, so we now have a place we can put these shared alter stateemnts